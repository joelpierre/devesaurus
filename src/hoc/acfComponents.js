import React from 'react';
import PropTypes from 'prop-types';

import TextBlock from '../components/organisms/TextBlock/TextBlock';
import InlineQuote from '../components/organisms/InlineQuote/InlineQuote';
import ImageBlock from '../components/organisms/ImageBlock/ImageBlock';
import CtaBanner from '../components/organisms/CtaBanner/CtaBanner';

const AcfComponents = ({ component, pageTheme }) => (
  <>
    {/**
     * @name Text Block Component
     * @prop module - object
     * @prop pageTheme - string
     */
    component.acf_fc_layout === 'text_block' && (
      <TextBlock
        data-test={`hoc-acf-${component.acf_fc_layout}`}
        module={component}
        pageTheme={pageTheme}
      />
    )}
    {/**
     * @name Image Block Component
     * @prop module - object
     * @prop pageTheme - string
     */
    component.acf_fc_layout === 'image_block' && (
      <ImageBlock
        data-test={`hoc-acf-${component.acf_fc_layout}`}
        module={component}
        pageTheme={pageTheme}
      />
    )}
    {/**
     * @name Inline Quote Component
     * @prop module - object
     * @prop pageTheme - string
     */
    component.acf_fc_layout === 'inline_quote' && (
      <InlineQuote
        data-test={`hoc-acf-${component.acf_fc_layout}`}
        module={component}
        pageTheme={pageTheme}
      />
    )}
    {/**
     * @name CTA Banner Component
     * @prop module - object
     * @prop pageTheme - string
     */
    component.acf_fc_layout === 'cta_banner' && (
      <CtaBanner
        data-test={`hoc-acf-${component.acf_fc_layout}`}
        module={component}
        pageTheme={pageTheme}
      />
    )}
  </>
);

AcfComponents.propTypes = {
  component: PropTypes.instanceOf(Object).isRequired,
  pageTheme: PropTypes.string.isRequired,
};

export default AcfComponents;
