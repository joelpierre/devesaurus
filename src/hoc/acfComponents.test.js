import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import acfComponents from './acfComponents';
import {
  checkProps,
  findByTestAttr,
  matchSnapshot,
} from '../utils/test-utilities';

const defaultProps = {
  component: {
    acf_fc_layout: 'text_block',
  },
  pageTheme: 'brand',
};

/**
 * Factory function to create a ShallowWrapper for the acfComponents component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<acfComponents {...setupProps} />);
};

describe('<acfComponents/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  xit('renders the acfComponents Component without errors', () => {
    const component = findByTestAttr(
      wrapper,
      `hoc-acf-${defaultProps.component.acf_fc_layout}`
    );

    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(acfComponents, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });
});
