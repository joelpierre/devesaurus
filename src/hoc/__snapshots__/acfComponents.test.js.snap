// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<acfComponents/> should match snapshot 1`] = `
<acfComponents
  component={
    Object {
      "acf_fc_layout": "text_block",
    }
  }
  pageTheme="brand"
/>
`;
